src/poetry/plugins/plugin_manager.py

```diff
@@ -41,11 +39,16 @@ def load_plugins(self, env: Env | None = None) -> None:
 
     def get_plugin_entry_points(
         self, env: Env | None = None
-    ) -> list[entrypoints.EntryPoint]:
-        entry_points: list[entrypoints.EntryPoint] = entrypoints.get_group_all(
-            self._group, path=env.sys_path if env else sys.path
-        )
-        return entry_points
+    ) -> list[metadata.EntryPoint]:
+        return [
+            ep
+            for ep in metadata.entry_points(group=self._group)
+            if env is None
+            or (
+                ep.dist is not None
+                and env.site_packages.find_distribution(ep.dist.name) is not None
+            )
```

Comment:
I would rewrite this to a normal loop. in its current form, this is barely readable.

Reply:
Split the conditional into another method for now.